### ISSUE TITLE
Encrypt backups?

### ISSUE THREAD

**Issue by TJM** [kenn/dumper#7]
How hard would it be to add the ability to encrypt backups? I am thinking that we could just override the "command" for the database provider and add  `| gpg ...` or whatever, but it would be nice if it was "supported" directly.

Tommy

**Comment by kenn** [comment]
That's a good idea. And you're right, you need to override `command` (and `filename`) if you want to do that now.

Some notes to myself:

- Use gpg's own compression with `--compress-algo` (ZIP by default)
- Use `gpg --yes --batch --passphrase=[passphrase]` for non-interactive processing

**Comment by TJM** [comment]
Alternatively, if you use `gpg` to encrypt with a "public key" you would not need a password at all, and as an added bonus, you would be able to ensure that no one could retrieve/read the backups unless they had the private key (pass phrase for the private key), where as the "shared key" model you have "noted" above would have the "key" in a config file on the server. 

~tommy

**Comment by kenn** [comment]
Yes, it's a trade-off between convenience and security - public key is definitely better for security but it adds operational complexity, so I'd opt for non-interactive / SSH-less workflow. Maybe it's possible to support both, though.
